aws/resource_aws_fsx_windows_file_system.go

```diff
@@ -174,6 +176,30 @@ func resourceAwsFsxWindowsFileSystem() *schema.Resource {
 					validation.StringMatch(regexp.MustCompile(`^[1-7]:([01]\d|2[0-3]):?([0-5]\d)$`), "must be in the format d:HH:MM"),
 				),
 			},
+			"deployment_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  fsx.WindowsDeploymentTypeSingleAz1,
+				ValidateFunc: validation.StringInSlice([]string{
+					fsx.WindowsDeploymentTypeMultiAz1,
+					fsx.WindowsDeploymentTypeSingleAz1,
+				}, false),
```

Comment:
Do you want to add `WindowsDeploymentTypeSingleAz2` as part of this PR? The option was added along with HDD support - https://github.com/terraform-providers/terraform-provider-aws/issues/12553 - but seems to have applicability  for SSD use cases.

Reply:
will add + test. it takes an hour to run and had to do a few tweaks 😢

Reply:
done.